src/instrumentation/podmutator.go

```diff
@@ -35,150 +45,250 @@ var (
 )
 
 type instPodMutator struct {
-	Client      client.Client
-	sdkInjector *sdkInjector
-	Logger      logr.Logger
-}
-
-type languageInstrumentations struct {
-	Java   *v1alpha1.Instrumentation
-	NodeJS *v1alpha1.Instrumentation
-	Python *v1alpha1.Instrumentation
-	DotNet *v1alpha1.Instrumentation
-	Php    *v1alpha1.Instrumentation
-	Ruby   *v1alpha1.Instrumentation
-	Go     *v1alpha1.Instrumentation
+	client                        client.Client
+	sdkInjector                   SdkInjector
+	secretReplicator              SecretReplicator
+	instrumentationLocator        InstrumentationLocator
+	getLicenseKeySecretFn         func(insts []*v1alpha2.Instrumentation) (string, error)
+	getLanguageInstrumentationsFn func(insts []*v1alpha2.Instrumentation) ([]*v1alpha2.Instrumentation, error)
```

Comment:
I don't think we need this field. 

You already the function 
`func GetLanguageInstrumentations(instCandidates []*v1alpha2.Instrumentation) ([]*v1alpha2.Instrumentation, error) {` 

In `Mutate`, you can call it `instCandidates, err := GetInstrumentations(ctx, ns, pod)`. 

This will also simplify the main function where you initialize `NewMutator`

Reply:
I removed it since I can mock the desired scenarios with the `instrumentationLocator`.